src/panel/direction/DirectionPanel.jsx

```diff
@@ -206,13 +259,26 @@ export default class DirectionPanel extends React.Component {
     this.setState(previousState => ({
       origin: previousState.destination,
       destination: previousState.origin,
+      originInputText: this.state.destinationInputText,
```

Comment:
`setState` is a potentially asynchronous operation, that's why when you set values based on other state values, you use this function form which gives safe access to the current state through it's first argument. So it's safer to do `previousState.destinationInputText`, like it's done above.
Here is a link to the explanation in the React doc. https://fr.reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

Reply:
k, can tou confirm me that I should also use previousState on lines 105 & 110 ?